test/models/user_test.rb

```diff
@@ -3,7 +3,61 @@
 require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
+  # rubocop:disable Metrics/BlockLength
+
   test 'from_omniauth' do
-    assert true
+    raw_auth_response = {
+      'provider' => 'github',
+      'uid' => '59379906',
+      'info' => {
+        'nickname' => 'andrewjumanca',
+        'email' => 'andrewjumanca@gmail.com',
+        'image' => 'https://avatars.githubusercontent.com/u/59379906?v=4'
+      },
+      'credentials' => {
+        'token' => '{\"p\":\"PO/SR+xSiR7NsxNdhwfiG8N8HqAAlXdqCQsJvdOfulxuSypzVAG+YQ==\",' \
+                   '\"h\":{\"iv\":\"VwxVWLF/5FO3Xt1K\",\"at\":\"WahZ6O4/aSdbTZXcgXmjnQ==\"}}'
+      },
+      'extra' => {
+        'raw_info' => {
+          'login' => 'andrewjumanca',
+          'avatar_url' => 'https://avatars.githubusercontent.com/u/59379906?v=4',
+          'url' => 'https://api.github.com/users/andrewjumanca',
+          'html_url' => 'https://github.com/andrewjumanca'
+        },
+        'all_emails' => [
+          { 'email' => '59379906+andrewjumanca@users.noreply.github.com' },
```

Comment:
You'll probably want to replace your personal emails with fake ones, especially since this is a public repo. Something like `test@example.com` would work fine.